comp/metadata/packagesigning/packagesigningimpl/yumsigning.go

```diff
@@ -0,0 +1,185 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package packagesigningimpl
+
+import (
+	"bufio"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+)
+
+const (
+	yumConf  = "/etc/yum.conf"
+	yumRepo  = "/etc/yum.repos.d/"
+	zyppConf = "/etc/zypp/zypp.conf"
+	zyppRepo = "/etc/zypp/repos.d/"
+)
+
+// getMainGPGCheck returns gpgcheck setting for [main] table
+func getMainGPGCheck(pkgManager string) bool {
+	repoConfig, _ := getRepoPathFromPkgManager(pkgManager)
+	gpgCheck, _ := parseRepoFile(repoConfig, false)
+	return gpgCheck
+}
+
+// getYUMSignatureKeys returns the list of keys used to sign RPM packages
+func getYUMSignatureKeys(pkgManager string) []SigningKey {
+	allKeys := make(map[SigningKey]struct{})
+	getKeysFromRepoFiles(allKeys, pkgManager)
+	getKeysFromRPMDB(allKeys)
+	var keyList []SigningKey
+	for keys := range allKeys {
+		keyList = append(keyList, keys)
+	}
+	return keyList
+}
+
+func getKeysFromRepoFiles(allKeys map[SigningKey]struct{}, pkgManager string) {
+	var gpgCheck bool
+	var gpgFiles []string
+	repoConfig, repoFilesDir := getRepoPathFromPkgManager(pkgManager)
+
+	// First parsing of the main config file
+	if _, err := os.Stat(repoConfig); os.IsExist(err) {
+		gpgCheck, gpgFiles = parseRepoFile(repoConfig, false)
+		for _, gpgFile := range gpgFiles {
+			decryptGPGFile(allKeys, gpgFile, "repo")
+		}
+	}
+	// Then parsing of the repo files
+	if _, err := os.Stat(repoFilesDir); os.IsExist(err) {
+		if files, err := os.ReadDir(repoFilesDir); err == nil {
+			for _, file := range files {
+				_, gpgFiles = parseRepoFile(file.Name(), gpgCheck)
+				for _, gpgFile := range gpgFiles {
+					decryptGPGFile(allKeys, gpgFile, "repo")
+				}
+			}
+		}
+	}
+}
+
+// getRepoPathFromPkgManager returns the path to the configuration file and the path to the repository files for RH or SUSE based OS
+func getRepoPathFromPkgManager(pkgManager string) (string, string) {
+	if pkgManager == "zypper" {
+		return zyppConf, zyppRepo
+	}
+	return yumConf, yumRepo
```

Comment:
We'll need to detect whether yum or dnf are present and use `/etc/dnf/dnf.conf` when DNF is used.